shared/teams/new-team/wizard/new-team-info.tsx

```diff
@@ -9,15 +9,34 @@ import {pluralize} from '../../../util/string'
 import {InlineDropdown} from '../../../common-adapters/dropdown'
 import {FloatingRolePicker} from '../../role-picker'
 import * as RouteTreeGen from '../../../actions/route-tree-gen'
+import * as RPCTypes from '../../../constants/types/rpc-gen'
 
 const NewTeamInfo = () => {
   const dispatch = Container.useDispatch()
   const nav = Container.useSafeNavigation()
 
   const teamWizardState = Container.useSelector(state => state.teams.newTeamWizard)
-  const teamNameTaken = false // TODO: get this live (Y2K-1524 and probably useRPC)
 
   const [name, setName] = React.useState(teamWizardState.name)
+  const [teamNameTakenError, setTeamNameTakenError] = React.useState<string | null>(null)
+  const [teamNameTaken, setTeamNameTaken] = React.useState(false)
+  const checkTeamNameTaken = Container.useRPC(RPCTypes.teamsUntrustedTeamExistsRpcPromise)
```

Comment:
also what is this magic :joy_cat: looks really cool

Reply:
It just calls the RPC promise and calls a callback when it has the result, but at the same time makes sure it doesn't call the result callback if the component is unmounted.